dagql/objects.go

```diff
@@ -521,9 +521,28 @@ func (r Instance[T]) call(
 		}
 	}
 
+	// field implementations can optionally return a wrapped Typed val that has
+	// a callback that should always run after the field is called
+	if postCallVal, ok := val.(*PostCallTyped); ok {
+		val = postCallVal.Typed
+		if postCallVal.PostCall != nil {
+			if err := postCallVal.PostCall(ctx); err != nil {
+				return nil, nil, fmt.Errorf("post-call error: %w", err)
+			}
+		}
+	}
+
 	return val, newID, nil
 }
 
+// PostCallTyped wraps a Typed value with an additional callback that
+// needs to be called after any value is returned, whether the value was from
+// cache or not
+type PostCallTyped struct {
+	Typed
```

Comment:
We might need/want this to also implement `Unwrap() Typed` so that things like telemetry middleware (`core.AroundFunc`) can still get to the underlying value

Reply:
Ah sure, I guess telemetry gets to it before the unwrapping happens in the current code, will fix, thanks!